Scrum 19 improve otp security features

alcarrillomunoz opened the pull request:
**Description**
This merge adds OTP security features, and makes some adjustments to data request messaging and routes changed GET to POST per team requests.

**Changes Made**
- installed OTP generator package
- created separate OTP data model, with 6 digit code, timestamps to track expiration, and reference to User to assign id (assignedTo); OTP is now created in OTP model when user requests password reset, time expiration is set for 10 minutes and is based on OTP updatedAt time; multiple users can now request reset password without overriding codes or access to reset function
- updated sendEmail function to add in href link for user with link to reset password (will need to update link per frontend team) 
- updated data responses for login/register/password reset controllers per backend team input; all now include status code, json response with data and message 
- updated reset password routes to be POST requests per front end team input

**Related Issues**
Will work with FE team to make sure new route set up connects properly.

**Additional Notes**
Testing was done via Postman and Ethereal email simulation program. Functions were testing correctly. 

**Merge Request Checklists**
Code follows project coding guidelines.
Documentation reflects the changes made.

Miminiverse commented on src/models/User.js:
I don't see this mentioned in description, the imageUrl I believe is for  user profile image? I'd suggest to give it a clearer name.

Miminiverse commented on src/controllers/resetPassword.js:
Some recommendations below: 

- Replace 60000 * 10 with a named constant for OTP expiration time to avoid magic number.
- After successful validation in OTP, invalidate the OTP.
- Add await for otp.save() and use a try catch if you can
- Move  line 59-61 to above 54, check if there's no new password first, then if it passes, check the user. That way it's not redundant to always check user.

alcarrillomunoz commented:
Thanks so much @Miminiverse! The following changes have been made: 
- User model imageURL was for profile picture, yes. I've changed the name to better reflect that. 
- added OTPexpirationTime named constant 
- invalidate OTP (set to null) after successful validation 
- added await to otp.save and used in try catch block with emailOTP function included 
- reordered reset password error handling to avoid redundancy

Miminiverse reviewed (approved):
Nice work!

alcarrillomunoz commented:
Additional changes made: 
- await added to user.save()
